transformer_engine/pytorch/csrc/common.cpp

```diff
@@ -97,6 +97,45 @@ transformer_engine::TensorWrapper makeTransformerEngineTensor(at::Tensor tensor)
   return makeTransformerEngineTensor(tensor.data_ptr(), shape, dtype);
 }
 
+std::tuple<std::vector<transformer_engine::TensorWrapper>, std::vector<std::vector<NVTETensor>>,
+           std::vector<NVTETensor*>, size_t, size_t>
+makeTransformerEngineTensor(std::vector<std::vector<at::Tensor>> at_tensor_lists) {
+  size_t num_lists = at_tensor_lists.size();
+
+  NVTE_CHECK(num_lists > 0, "List of tensors is empty.");
+
+  size_t num_tensors = at_tensor_lists[0].size();
+
+  std::vector<std::vector<NVTETensor>> nvte_tensor_lists;
+  std::vector<NVTETensor*> nvte_tensor_list_ptrs;
+  std::vector<transformer_engine::TensorWrapper> tensorWrappers;
+  nvte_tensor_lists.reserve(num_lists);
+  nvte_tensor_list_ptrs.reserve(num_lists);
+  tensorWrappers.reserve(num_lists * num_tensors);
+
+  for (const auto& at_list : at_tensor_lists) {
+    NVTE_CHECK(at_list.size() == num_tensors, "Wrong number of tensors");
+    std::vector<NVTETensor> te_list;
+    te_list.reserve(num_tensors);
+
+    for (const auto& at_tensor : at_list) {
+      transformer_engine::TensorWrapper tensorWrapper = makeTransformerEngineTensor(at_tensor);
+      NVTETensor te_tensor = tensorWrapper.data();
+      tensorWrappers.push_back(std::move(tensorWrapper));
+      te_list.push_back(std::move(te_tensor));
```

Comment:
Move operators don't seem right here since they can leave the original variable in an undefined state. If I were reading this for the first time, I would be uncertain if `te_tensor` is invalid after `tensorWrapper` has been moved. In practice it's fine since `NVTETensor` is a POD, so it will actually copy instead of move, but it's not clean.
```suggestion
      tensorWrappers.push_back(makeTransformerEngineTensor(at_tensor));
      te_list.push_back(tensorWrappers.back().data());
```